### DISCUSSION TITLE
The state of the MoinMoin project

### DISCUSSION THREAD

**Discussion by ant-222** [moinwiki/moin#1717] [General]
Dear maintainers,  
I am writing on behalf of the *FreeBSD* community, whose [wiki](https://wiki.freebsd.org/) is powered by *MoinMoin* 1.9. The latest stable release having come out more than 3½ years ago and depeding on an old major version of Python, we are concerned about the future of *MoinMoin* and are considering migration to some other,  steadily maintained, wiki engine.

Before we make such a decision, however, we should like to ask you about the state of the project: Is a new production release planned in the observable future? Do you have an estimate when it may be out?

**Comment by RogerHaase** [comment]
No estimate on date for stable production release.

Hoping to release 2.0.0b1 in August.

There has been little feedback on 2.0.0a1 release. More wiki admins need to test the migration process, test the resulting moin2 wiki, and tell us whether moin2 works for you. More developers fixing bugs would also help.

**Comment by UlrichB22** [comment]
The beta 2.0.0b1 has been released, thanks to Roger. 

For anyone in Germany who is interested in the moin project, there will be an opportunity to talk about the project next week on FrOSCon in St. Augustin. Please see [froscon.org](https://froscon.org/) and [FrOSCon2024 program](https://programm.froscon.org/2024/schedule/1.html#1).

**Comment by malemburg** [comment]
Hi there, I am one of the maintainers of the wiki.python.org wiki which runs moin 1.9.

It's a fairly large wiki (31k pages, 221k users) and we are often struggling with issues related to using too many inodes, searches hitting the server hard and the need to do regular cleanup in order to keep things running.

Since moin2 seems to be coming along nicely, I am wondering whether we should start trying to migrate to moin2 at this point, or still wait for a few more betas. Any ideas ?

**Reply by UlrichB22** [reply]
IMO it is the right time to start as the migration is time consuming anyway. Let's find out what the difficulties are in the case of your wiki. I will contact you and we can discuss how I can support you.

**Reply by sebix** [reply]
Be aware of https://github.com/moinwiki/moin/issues/1725

**Reply by malemburg** [reply]
Thanks, @UlrichB22 and @sebix

We will have to do a test installation to check issues such as these. Those 31k pages will likely get trimmed down a bit, since we do have quite a number of spam pages in there, but I doubt that we'll get below 10k pages.

**Reply by malemburg** [reply]
Correction for the numbers: We have 3400+ pages and 47k users. 

I was previously looking at a backup copy which doesn't remove old entries (for various reasons). Since we run maintenance scripts on the server on a daily basis, e.g. to clean up pages and sessions, the numbers were (much) higher.

Still, a sizable installation.

**Comment by UlrichB22** [comment]
On August 16 and 17, interested people in Germany will once again have the opportunity to meet at FrOSCon and discuss about Moin2. I would also be happy to assist with the test migration from Moin 1.9 to 2.0 and with configuring and customizing the interface for public wikis. 

FrOSCon takes place in St. Augustin near Bonn. Further information can be found at [froscon.org](https://froscon.org/en/).

**Reply by ThomasWaldmann** [reply]
Will you do a talk or a lightning talk?

At some conferences, there is also the possibility to do self-organised sessions - not sure if that is the case for FrOSCon.

**Reply by ThomasWaldmann** [reply]
BTW, maybe open a new ticket about this for better visibility?

**Reply by UlrichB22** [reply]
> Will you do a talk or a lightning talk?

We have a dedicated Devroom for Moinmoin Wiki. There will be two talks for people who have no experience with moin or moin2, as well as two workshops on installing a moin wiki or migrating to moin2.

**Reply by ThomasWaldmann** [reply]
Cool! :-) Sadly, I can't come.

**Reply by malemburg** [reply]
Me neither 😞

**Comment by cthart** [comment]
Is there any prediction when 2.0 will be ready? Like others commenting here, we have a MoinMoin 1.9 wiki and also have the upgrade-or-migrate dilemma. Ours isn't huge though.

**Reply by rikvanriel** [reply]
I've decided to just bite the bullet, and migrate some wikis to moin 2.0. I have linux-mm.org and spamikaze.org on moin 2.0 right now.

I will open pull requests as I run into issues.

**Reply by roland-ruedenauer** [reply]
@rikvanriel Great to hear about that. Also thanks for the fixes you've submitted so far.

Regarding your Moin2 wikis: you might have missed setting a default max age for files served via Flask's send_from_directory in your wikiconfig.py

https://github.com/moinwiki/moin/blob/87f37e8dcbc2e6ee7e8e7f128b46a69c8c6661bb/src/moin/config/wikiconfig.py#L300

This should help avoiding unnecessary traffic.

**Reply by rikvanriel** [reply]
Thank you for that config setting! I just enabled it across the wikis.

I will send in more PRs as I run into other issues. Thank you for merging things so quickly.

**Comment by UlrichB22** [comment]
@cthart, have you already run a test migration? Do you think there are any issues?

**Reply by cthart** [reply]
Not yet. I will give it a shot within the next week or so.
